src/Command/Geo/SyncZonesCommand.php

```diff
@@ -0,0 +1,156 @@
+{
+    private const CLASSES = [
+        Country::class,
+        Region::class,
+        Department::class,
+        District::class,
+        Canton::class,
+        CityCommunity::class,
+        City::class,
+        ForeignDistrict::class,
+        ConsularDistrict::class,
+    ];
+
+    protected static $defaultName = 'app:geo:sync-zones';
+
+    /**
+     * @var EntityManagerInterface
+     */
+    private $em;
+
+    /**
+     * @var SymfonyStyle
+     */
+    private $io;
+
+    /**
+     * @var Collection
+     */
+    private $entities;
+
+    public function __construct(EntityManagerInterface $em)
+    {
+        $this->em = $em;
+
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this
+            ->setDescription('Create missing zone and link it to its parents')
+            ->addOption('dry-run', null, InputOption::VALUE_NONE, 'Execute the algorithm but will not persist in database.')
+        ;
+    }
+
+    protected function initialize(InputInterface $input, OutputInterface $output): void
+    {
+        $this->io = new SymfonyStyle($input, $output);
+        $this->entities = new ArrayCollection();
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $this->io->title('Start syncing zones');
+
+        $zonables = $this->findZonables();
+
+        $this->io->section('Transforming source entities into zones');
+
+        $this->io->progressStart(\count($zonables));
+
+        foreach ($zonables as $zonable) {
+            $this->zoneableAsZone($zonable);
+            $this->io->progressAdvance();
+        }
+
+        $this->io->progressFinish();
+
+        $summary = new Summary($this->io);
+        $summary->run($this->entities);
+
+        $persister = new Persister($this->io, $this->em);
+        $persister->run($this->entities, $input->getOption('dry-run'));
+
+        return 0;
+    }
+
+    /**
+     * @return array|ZoneableInterface[]
+     */
+    private function findZonables(): iterable
+    {
+        $grouped = [];
+
+        $this->io->section('Fetch zones from sources');
+
+        foreach (self::CLASSES as $class) {
+            $this->io->write(sprintf(' // "%s"', $class));
+
+            $entities = $this->em
+                ->createQueryBuilder()
+                ->select(['x'])
+                ->from($class, 'x')
+                ->getQuery()
+                ->getResult()
+            ;
+
+            $this->io->writeln(sprintf(' ... found %d', \count($entities)));
+
+            $grouped[] = $entities;
+        }
+
+        $entities = array_merge(...$grouped);
+
+        $this->io->comment(sprintf('Total %d', \count($entities)));
+
+        return $entities;
+    }
+
+    private function zoneableAsZone(ZoneableInterface $zoneable): Zone
+    {
+        $repository = $this->em->getRepository(Zone::class);
+
+        $type = $repository->getTypeByZonable($zoneable);
```

Comment:
it will be nice to have `$zoneable->getType()` instead, wdyt?

Reply:
`$repository->getTypeByZonable()` has been removed, I've added `$zoneable->getZoneType()`